### PR TITLE
Upload: better async upload queue

### PR DESCRIPTION
**Pull request by distantnative**
Hi @afbora, would love to get your help on this, testing all kinds of upload scenarios and if they are still working. Thanks!

### CODE REVIEW

**Comment by afbora** [comment]
Uploading avatar throws following error:

> Uncaught TypeError: this.start is not a function at HTMLInputElement.<anonymous> (index.min.js:1:379743)

https://github.com/getkirby/kirby/blob/v4/enhancement/upload-queue/panel/src/panel/upload.js#L179

**Comment by afbora** [comment]
~I don't know following issue is related with this PR: Creating thumbnail issue with replace while in lock state.~

Edit: Moved to https://github.com/getkirby/kirby/issues/5566

File: panel/src/panel/upload.js [hunk 1]
```diff
@@ -164,7 +176,7 @@ export default (panel) => {
 					// if upload should start immediately
 					this.set(options);
 					this.select(event.target.files);
-					this.start();
+					this.submit();
```
Comment (afbora) [inline panel/src/panel/upload.js:179]:
```js
await this.submit();
```

It doesn't have to be like this, does it?

Comment (distantnative) [reply panel/src/panel/upload.js:179]: I don't think so, because we want `this.input.remove()` to happen directly even while the upload is running.

**Comment by distantnative** [comment]
@afbora Not sure the thumbnail issue is related. Could be a general v4 thing or even broken in v3?

**Comment by afbora** [comment]
@distantnative You're right. That issue exists already 3.9.6. I moved to https://github.com/getkirby/kirby/issues/5566

**Review by afbora — APPROVED** [review]
LGTM. All works great for me 👍

**Pull request merged**